### PR TITLE
Add: Daily data key collect and page_archives route

### PR DESCRIPTION
**Pull request by nEEtdo0d**
CN, JP, and TW will need to update their argparser and replace DATA_KEY_COLLECTED, optionally may replace DATA_KEY_COLLECT

### CODE REVIEW

File: module/war_archives/data_key_collect.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            if self.appear(DATA_KEY_COLLECTED):
+                logger.info('Shikikan has already collected today\'s data key')
+                break
+
+            current_count = OCR_CURRENT_COUNT.ocr(self.device.image)
+            max_count = OCR_MAX_COUNT.ocr(self.device.image)
+            logger.info(f'Data Key Status: {current_count} / {max_count}')
+            if current_count < max_count:
+                self.device.click(DATA_KEY_COLLECT)
+                result = self._collect_confirm()
```
Comment (LmeSzinc) [inline module/war_archives/data_key_collect.py:69]: A simple click on DATA_KEY_COLLECT may not stable.
Logic should be
```
for n in range(3):
    if self.appear(DATA_KEY_COLLECTED):
        return True
    current, remain, total = OCR_DATA_KEY.ocr(self.device.image)
    if remain <= 0:
        return True

    self.device.click(DATA_KEY_COLLECT)
    self._collect_confirm()
    continue

logger.warning('xxxxx')
return False

**Review by LmeSzinc — COMMENTED** [review]
DataKeyCollect should be inherited and called by Reward, not Daily.
Function names in class DataKeyCollect have a prefix, such as `data_key_` or `achieves_`.

File: module/war_archives/data_key_collect.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nEEtdo0d) [reply module/war_archives/data_key_collect.py]: Converted from Digit to DigitCounter

**Comment by nEEtdo0d** [comment]
- Moved DataKey from daily to reward
- Prepend methods with `data_key`
- Converted Digit to DigitCounter

**Pull request merged**